openmdao/core/implicitcomponent.py

```diff
@@ -71,6 +71,13 @@ def _solve_nonlinear(self):
             else:
                 return result
 
+    def _guess_nonlinear(self):
+        """
+        Provide initial guess for states.
+        """
+        with self._unscaled_context(outputs=[self._outputs], residuals=[self._residuals]):
+            self.guess_nonlinear(self._inputs, self._outputs, self._residuals)
```

Comment:
Technically you could put Newton on a component, but I don't know how that would work unless you manually call the solve in your solve_nonlinear, and if you do that, you will probably end up inside of nested calls of the context manager ... don't know if we check for that and prevent multiplying by the units scales again. So, not sure it even works in practice.